docs/utilities/idempotency.md

```diff
@@ -215,12 +217,63 @@ Resources:
     see 1WCU and 1RCU. Review the [DynamoDB pricing documentation](https://aws.amazon.com/dynamodb/pricing/) to
     estimate the cost.
 
+#### Using Redis
+
+##### Redis resources
+
+You need an existing Redis service before setting up Redis as the persistent storage layer provider. You can also use Redis compatible services like [Amazon ElastiCache for Redis](https://aws.amazon.com/elasticache/redis/) or [Amazon MemoryDB for Redis](https://aws.amazon.com/memorydb/) as persistent storage layer provider.
+!!! tip "Tip:No existing Redis service?"
+    If you don't have an existing Redis service, we recommend using DynamoDB as persistent storage layer provider. DynamoDB does not require a VPC deployment and is easier to configure and operate.
+
+If you want to connect to a Redis cluster instead of a Standalone server, you need to enable Redis cluster mode by setting an AWS Lambda
+environment variable `REDIS_CLUSTER_MODE` to `true`
+In the following example, you can see a SAM template for deploying an AWS Lambda function by specifying the required environment variable.
+
+!!! warning "Warning: Large responses with Redis persistence layer"
+When using this utility with Redis your function's responses must be smaller than 512MB.
+Persisting larger items cannot might cause exceptions.
+
+```yaml hl_lines="9" title="AWS Serverless Application Model (SAM) example"
+Resources:
+  IdempotencyFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: Function
+      Handler: helloworld.App::handleRequest
+      Environment:
+        Variables:
+          REDIS_CLUSTER_MODE: "true"
+```
+##### VPC Access
+Your AWS Lambda Function must be able to reach the Redis endpoint before using it for idempotency persistent storage layer. In most cases you will need to [configure VPC access](https://docs.aws.amazon.com/lambda/latest/dg/configuration-vpc.html) for your AWS Lambda Function. Using a public accessible Redis is not recommended.
+
+!!! tip "Amazon ElastiCache for Redis as persistent storage layer provider"
+    If you intend to use Amazon ElastiCache for Redis for idempotency persistent storage layer, you can also reference [This AWS Tutorial](https://docs.aws.amazon.com/lambda/latest/dg/services-elasticache-tutorial.html).
+
+```yaml hl_lines="7-12" title="AWS Serverless Application Model (SAM) example"
+Resources:
+  IdempotencyFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: Function
+      Handler: helloworld.App::handleRequest
+      VpcConfig: # (1)!
+        SecurityGroupIds: # (2)!
+          - sg-{your_sg_id}
+        SubnetIds: # (3)!
+          - subnet-{your_subnet_id_1}
+          - subnet-{your_subnet_id_2}
+```
+1. Replace the Security Group ID and Subnet ID to match your Redis' VPC setting.
+2. The security group ID or IDs of the VPC where the Redis deployment is configured.
+3. The subnet IDs  of the VPC where the Redis deployment is configured.
```

Comment:
```suggestion
3. The subnet IDs of the VPC where the Redis deployment is configured.
```

what does "where the Redis deployment is configured" mean? isn't it "where Redis is deployed" ?